docs/man/docker-build.1.md

```diff
@@ -41,6 +43,22 @@ as context.
 **-f**, **--file**=*PATH/Dockerfile*
    Path to the Dockerfile to use. If the path is a relative path then it must be relative to the current directory. The file must be within the build context. The default is *Dockerfile*.
 
+**--build-env**=*environment*
+   Set build-time environment variables. This option allows you to specify
+values of the variables that are available for expansion/substitution in the
+Dockerfile primitives like ADD, COPY etc, without an explicit prior definition by
+the ENV primitive. The build-time environment variable are also passed as environment
+context for the command(s) that will be executed as part of 'RUN' primitive
+of Dockerfile, if there is no explicit prior definition by the ENV primitive.
+Note that these variables get persisted in the intermediate and final docker
+images, if they result in an expansion or substitution as part of Dockerfile primitives.
+However, they are not persisted when used as environment context for command(s)
+run as part of `RUN`. This gives the flexibility to build an image by passing host specific
+environment variables (like http_proxy) without affecting portability of generated image.
```

Comment:
This might be confusing because the order is backwards to what I think people would expect.
By that I mean, let's talk about the normal/easy case first.
```
Normally, these variables are not persisted in the resulting Docker image. This gives
the flexibility to build an image by passing host specific environment variables (like 
http_proxy) that will be used on the RUN commands without affecting portability
of the generated image.
However, as with any variable, they can be persisted in the final image if they are used in an
ENV command (e.g. ENV myName=$myName will save myName in the image).
```